File: recyclerview/app/src/main/java/com/example/cntt/myapplication/AdapterStudent.java
package com.example.cntt.myapplication;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

/**
 * Created by CNTT on 3/22/2018.
 */

public class AdapterStudent extends RecyclerView.Adapter<AdapterStudent.ViewHolder> {
    List<Student> listStudent;
    Click click;

    public AdapterStudent(List<Student> listStudent, Click click) {
        this.listStudent = listStudent;
        this.click = click;

    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        Student student=listStudent.get(position);
        holder.txt_name.setText(student.getName());
        holder.txt_year_old.setText(student.getYear_old()+"");


    }

    @Override
    public int getItemCount() {
        return listStudent.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
         TextView txt_name, txt_year_old;

        public ViewHolder(View itemView) {
            super(itemView);
            txt_name = itemView.findViewById(R.id.txt_name);
            txt_year_old = itemView.findViewById(R.id.txt_yead_old);
            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    click.onClickView(getPosition());
                }
            });
        }

    }
}

File: jetpack/Databinding/app/src/main/java/com/audiovideoplayer/databinding/adapter/AdapterStudent.kt
package com.audiovideoplayer.databinding.adapter

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.audiovideoplayer.databinding.R
import com.audiovideoplayer.databinding.interfaces.ItemClick
import kotlinx.android.synthetic.main.item_student.view.*

class AdapterStudent(var list: ArrayList<String>,var itemClick: ItemClick) : RecyclerView.Adapter<AdapterStudent.ViewHolder>() {
    override fun onCreateViewHolder(p0: ViewGroup, p1: Int): ViewHolder {
        return ViewHolder(LayoutInflater.from(p0.context).inflate(R.layout.item_student, p0, false))
    }

    override fun getItemCount(): Int {
        return list.size
    }

    override fun onBindViewHolder(p0: ViewHolder, p1: Int) {
        p0.itemView.txtName.text=list[p1]
        p0.itemView.setOnClickListener {itemClick.itemClick(p0.adapterPosition) }
    }

    class ViewHolder(itemview: View) : RecyclerView.ViewHolder(itemview)

}